rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -44,9 +45,21 @@ Node::Node(std::string node_name, context::Context::SharedPtr context)
 : name_(node_name), context_(context),
   number_of_subscriptions_(0), number_of_timers_(0), number_of_services_(0)
 {
-  node_handle_ = rmw_create_node(name_.c_str());
+  // Initialize node handle shared_ptr with custom deleter.
+  node_handle_.reset(rmw_create_node(name_.c_str()), [ = ](rmw_node_t * node) {
```

Comment:
Why does the lambda takes a pointer to an `rmw_node_t` instead of a shared pointer?